src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,79 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int INITIAL_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        Node node = new Node(key, value, null);
+        int index = (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        }
+        for (Node<K, V> i = table[index]; i != null; i = i.next) {
+            if (Objects.equals(i.key, node.key)) {
+                i.value = (V) node.value;
+                break;
+            }
+            if (i.next == null) {
+                i.next = node;
+                size++;
+                break;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
+        for (Node<K, V> i = table[index]; i != null; i = i.next) {
+            if (Objects.equals(i.key, key)) {
+                return i.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
```

Comment:
You can remove the field "next" from the constructor. Because where you create object Node, in this place you transfer null. It is field number 20